iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/node/WALNode.java

```diff
@@ -833,6 +833,21 @@ public void skipTo(long targetIndex) {
             targetIndex,
             targetIndex);
       }
+
+      if (itr != null
+          && itr.hasNext()
+          && insertNodes.get(itr.nextIndex()).getSearchIndex() <= targetIndex
+          && targetIndex <= insertNodes.get(insertNodes.size() - 1).getSearchIndex()) {
```

Comment:
How about changing the type of insertNodes to LinkedList, so you can use `insertNodes.getLast().getSearchIndex()` here